npm/private/npm_translate_lock_helpers.bzl

```diff
@@ -0,0 +1,52 @@
+"""Starlark helpers for npm_translate_lock."""
+
+load("@bazel_skylib//lib:new_sets.bzl", "sets")
+
+# Adapted from https://github.com/bazelbuild/bazel/issues/3901#issuecomment-421318536
+def _directory_path(rctx, label, user_workspace_label):
+    """Return the absolute path to a directory referred to by an absolute label in the user's workspace"""
+    label = str(label)
+
+    split = label.split("/", 1)
+    split = split[1].split(":", 1)
+    dir_path = split[0]
+    dir_name = split[1]
+    if dir_path == "/":
+        path = "/" + dir_name
+    else:
+        path = dir_path + "/" + dir_name
+
+    return rctx.path(
+        str(rctx.path(user_workspace_label)).rsplit("/", 1)[0] +
+        path,
+    )
+
+def _verify_patches(rctx):
+    if rctx.attr.verify_patches and rctx.attr.patches != None:
+        patches_folder_path = _directory_path(rctx, rctx.attr.verify_patches, rctx.attr.pnpm_lock)
+
+        all_patch_files = sets.make(patches_folder_path.readdir())
```

Comment:
lets filter on `.patch` and `.diff` files? perhaps make this configurable as well will with a `verify_patches_extensions` attribute that takes a list of extensions and defaults to `[".patch", "diff"]`?